### ISSUE TITLE
Enums are not namespaced properly with near-operation-file

### ISSUE THREAD

**Issue by alexpyoung** [dotansimha/graphql-code-generator#2398]
**Describe the bug**
 

When using the `near-operation-file` preset, generated `enum` types are not properly namespaced using the base path type file.

**To Reproduce**
1. `git clone git@github.com:alexpyoung/codegen-foo.git`
2. `yarn install && yarn compile`

1. My GraphQL schema:

```graphql

type Foo {
    bar: Bar!
}

enum Bar {
    FOO
    BAR
    BAZ
}

type Query {
    foo: Foo!
}

schema {
    query: Query
}
```

2. My GraphQL operations:

```graphql
query GetFoo {
    foo {
        bar
    }
}
```

3. My `codegen.yml` config file:

```yml
overwrite: true
schema: "./schema.graphql"
documents: "./operations/*.gql"
generates:
  ./types.ts:
    plugins:
      - typescript
    config:
      avoidOptionals: true
      immutableTypes: true
      preResolveTypes: true
  ./operations/:
    preset: near-operation-file
    presetConfig:
      extension: .graphql.tsx
      baseTypesPath: ../types
    plugins:
      - typescript-operations
      - typescript-react-apollo
    config:
      hooksImportFrom: "@apollo/react-hooks"
      immutableTypes: true
      preResolveTypes: true
      withHOC: false
      withHooks: true
```

**Expected behavior**
TypeScript transpilation should succeed

**Actual behavior**
```
operations/foo.graphql.tsx:1:1 - error TS6133: 'Types' is declared but its value is never read.

1 import * as Types from '../types';
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

operations/foo.graphql.tsx:12:119 - error TS2304: Cannot find name 'Bar'.

12 export type GetFooQuery = { readonly __typename?: 'Query', readonly foo: { readonly __typename?: 'Foo', readonly bar: Bar } };
                                                                                                                         ~~~


Found 2 errors.

error Command failed with exit code 1.
```

**Environment:**

- OS: macOS 10.14.3

**Additional context**
https://github.com/alexpyoung/codegen-foo

**Comment by dotansimha** [comment]
Fixed in: https://github.com/dotansimha/graphql-code-generator/pull/2458

**Comment by dotansimha** [comment]
Fixed in 1.7.0
